### ISSUE TITLE
maven uploadArchives is not uploading sources.jar or javadoc.jar but does upload the signed files

### ISSUE THREAD

**Issue by kevin-j-smith** [gradle/gradle#3850]
I am working on a gradle plugin for my company.   It dynamically creates a uploadArchives closure provided some artifacts that I am wanting uploaded and a gradle extension that defined common company information.  These files are built in a separate process and during the upload process they are needing signed.   After upload I am seeing the following files:
artifact-version-javadoc.asc
artifact-version-javadoc.asc.md5
artifact-version-javadoc.asc.sha1
artifact-version-sources.asc
artifact-version-sources.asc.md5
artifact-version-sources.asc.sha1
artifact-version.pom
artifact-version.pom.asc
artifact-version.pom.asc.md5
artifact-version.pom.asc.sha1
artifact-version.tar
artifact-version.tar.asc
artifact-version.tar.asc.md5
artifact-version.tar.asc.sha1
artifact-version.tar.md5
artifact-version.tar.sha1

As you can see I am missing 6 files:
artifact-version.javadoc.jar
artifact-version.sources.jar
**and the accompanying md5 and sha1 files

Here is the plugin's code that I am using to dynamically create this behavior. 
`
final ConfigurationContainer configurations = project.getConfigurations();
def configuration = configurations.create(CONFIGURATION_NAME)

def extension = project.extensions[EXTENSION_NAME]
File buildDirectory = extension.buildDirectory
extension.artifacts.each() { File artifact ->
  project.artifacts.add(CONFIGURATION_NAME, artifact,{
    name = project.artifactName
    classifier = artifact.getClassifier()  //metaClass was added to parse the artifact name to get the classifier
    extension = artifact.getExtension(true) //metaClass was added to parse the artifact name to get the extension
  })
}

def emptySourceFile = new File(buildDirectory,'source.txt');
def emptySourceJar = new File(buildDirectory,"sources.jar");
emptySourceFile << "Some fun text. ${project.version}"['jar','cf',emptySourceJar,emptySourceFile.getName()].execute(null,buildDirectory).waitForOrKill(1000);
project.artifacts.add(CONFIGURATION_NAME, emptySourceJar, {
  classifier = "sources"
  type = "sources"
})

def emptyJavadocFile = new File(buildDirectory,'javadoc.txt');
def emptyJavadocJar = new File(buildDirectory,"javadoc.jar");
emptyJavadocFile << "Some fun text. ${project.version}"
['jar','cf',emptyJavadocJar,emptyJavadocFile.getName()].execute(null,buildDirectory).waitForOrKill(1000);
project.artifacts.add(CONFIGURATION_NAME, emptyJavadocJar, {
  classifier = "javadoc"
  type = "javadoc"
})

project.uploadArchives {
  repositories {
    mavenDeployer {
      beforeDeployment { MavenDeployment deployment -> project.signing.signPom(deployment)}
      repository(url: deployRepo) {
        authentication(userName: project.ossrhUsername, password: project.ossrhPassword)
      }
      configuration = project.configurations.getByName(CONFIGURATION_NAME)

      pom.project {
        name project.archivesBaseName
        description project.extensions[EXTENSION_NAME].description
        url project.extensions[EXTENSION_NAME].url
        scm {
          url project.extensions[EXTENSION_NAME].scm.url
          connection project.extensions[EXTENSION_NAME].scm.connection
          developerConnection project.extensions[EXTENSION_NAME].scm.developerConnection
        }
        licenses {
          license {
            name project.extensions[EXTENSION_NAME].license.name
            url project.extensions[EXTENSION_NAME].license.url
          }
        }
        developers {
          developer {
            name project.extensions[EXTENSION_NAME].developer.name
            email project.extensions[EXTENSION_NAME].developer.email
            organization project.extensions[EXTENSION_NAME].developer.organization
            organizationUrl project.extensions[EXTENSION_NAME].developer.organizationUrl
          }
        }
      }
    }
  }
}
`
In the debug output I see the PUT requests which don't have source.jar or javadoc.jar, but has their signed files.  

12:02:13.830 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.tar.asc HTTP/1.1
12:02:51.963 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.tar.asc.sha1 HTTP/1.1
12:02:55.397 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.tar.asc.md5 HTTP/1.1
12:02:58.438 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.pom HTTP/1.1
12:02:58.705 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.pom.sha1 HTTP/1.1
12:02:58.949 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.pom.md5 HTTP/1.1
12:03:01.857 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version-sources.jar.asc HTTP/1.1
12:03:02.200 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version-sources.jar.asc.sha1 HTTP/1.1
12:03:02.444 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version-sources.jar.asc.md5 HTTP/1.1
12:03:02.675 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version-javadoc.jar.asc HTTP/1.1
12:03:02.901 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version-javadoc.jar.asc.sha1 HTTP/1.1
12:03:03.116 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version-javadoc.jar.asc.md5 HTTP/1.1
12:03:03.341 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.tar HTTP/1.1
12:03:03.617 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.tar.sha1 HTTP/1.1
12:03:03.849 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.tar.md5 HTTP/1.1
12:03:04.084 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.pom.asc HTTP/1.1
12:03:04.353 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.pom.asc.sha1 HTTP/1.1
12:03:04.582 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/version/artifact-version.pom.asc.md5 HTTP/1.1
12:03:10.456 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/maven-metadata.xml HTTP/1.1
12:03:11.591 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/maven-metadata.xml.sha1 HTTP/1.1
12:03:11.808 [DEBUG] [org.apache.http.impl.conn.DefaultClientConnection] Sending request: PUT /service/local/staging/deploy/maven2/group/artifact/maven-metadata.xml.md5 HTTP/1.1

### Expected Behavior
I expect to see all of the files that are defined in my extension, the empty source and javadoc jar files, the signed files, the hash files and the pom file all uploaded to the maven repository.  I also expect to see a PUT statement for sources.jar, sources.jar.md5, sources.jar.sha1, javadoc.jar, javadoc.jar.md5, and javadoc.jar.sha1

### Current Behavior
Missing sources.jar and javadoc.jar files that were added to the artifacts list of the configuration.  

### Context
I cannot publish my artifact without sources.jar and javadoc.jar
These are things needed for oss.sonatype to publish to maven central

### Steps to Reproduce (for bugs)
Submitted code above.   With that in a Project<Plugin> class I can run "gradle build.gradle uploadArchives"

### Your Environment
gradle --version

------------------------------------------------------------
Gradle 4.0.2
------------------------------------------------------------

Build time:   2017-07-26 15:04:56 UTC
Revision:     108c593aa7b43852f39045337ee84ee1d87c87fd

Groovy:       2.4.11
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.7.0 (IBM Corporation 2.7)
OS:           Linux 3.10.0-693.2.2.el7.x86_64 amd64

**Comment by oehme** [comment]
The `uploadArchives` task only uploads what's in the `archives` configuration. You are adding your source/javadoc to a different configuration though.

**Comment by kevin-j-smith** [comment]
@oehme I did tell uploadArchives to use the configuration that I created. 
configuration = project.configurations.getByName(CONFIGURATION_NAME)

 That can be seen by all of the other files being uploaded.

**Comment by kevin-j-smith** [comment]
@oehme.  I did not mention that I have a list of files that I am uploading with different classifiers and have put those files in artifacts under my defined configuration.  

artifact.version.linux-x86_64.extension
...all of the signed files and md5 and sha1 of those
artifact.version.windows64.extension
...all of the signed files and md5 and sha1 of those
...classifier = arm64
...classifier = ...and so on and so on

Those files are found in maven with no issues.  The only files that I have added to my defined configuration and are not in maven are sources.jar and javadoc.jar.  Even though the signed files are there.
